### PR TITLE
feat(obj): add force_destroy option to bucket resource

### PR DESCRIPTION
**Pull request by Codelax**

### CODE REVIEW

File: scaleway/resource_object_bucket.go [hunk 1]
```diff
@@ -275,6 +281,13 @@ func resourceScalewayObjectBucketDelete(ctx context.Context, d *schema.ResourceD
 		return diag.FromErr(err)
 	}
 
+	if d.Get("force_destroy").(bool) {
+		err = deleteS3Objects(ctx, s3Client, bucketName)
```
Comment (Monitob) [inline scaleway/resource_object_bucket.go:285]: Hello @Codelax, please consider the versions if there are

File: scaleway/resource_object_bucket.go [hunk 2]
```diff
@@ -275,6 +281,13 @@ func resourceScalewayObjectBucketDelete(ctx context.Context, d *schema.ResourceD
 		return diag.FromErr(err)
```
Comment (Monitob) [inline scaleway/resource_object_bucket.go:281]: The delete is a new topic for this resource. Please check if the error is implemented as [here](https://github.com/hashicorp/terraform-provider-aws/blob/26f2311f79416f5d37c439288a7287a3b6a56c26/internal/service/s3/bucket.go#L1288).

Comment (Monitob) [reply scaleway/resource_object_bucket.go:281]: Please check if bucket exist like [this](https://github.com/hashicorp/terraform-provider-aws/blob/26f2311f79416f5d37c439288a7287a3b6a56c26/internal/service/s3/bucket.go#L1284)

File: scaleway/resource_object_bucket.go (hunk 1 above)
Comment (Monitob) [reply scaleway/resource_object_bucket.go:285]: You can check a similar implementation [here](https://github.com/hashicorp/terraform-provider-aws/blob/26f2311f79416f5d37c439288a7287a3b6a56c26/internal/service/s3/bucket.go#L1305)

File: scaleway/helpers_object.go [hunk 3]
```diff
@@ -195,3 +196,35 @@ func expandBucketCORS(rawCors []interface{}, bucket string) []*s3.CORSRule {
...
+func deleteS3Objects(ctx context.Context, conn *s3.S3, bucketName string) error {
+	var err error
+	listInput := &s3.ListObjectsInput{
+		Bucket: scw.StringPtr(bucketName),
+	}
+	listErr := conn.ListObjectsPagesWithContext(ctx, listInput, func(page *s3.ListObjectsOutput, lastPage bool) bool {
+		for _, object := range page.Contents {
+			err = deleteS3Object(conn, bucketName, object)
+			if err != nil {
+				err = fmt.Errorf("failed to delete S3 object: %s", err)
```
Comment (Monitob) [inline scaleway/helpers_object.go:217]: Add the error definition on the constants with a description

**Review by Monitob — CHANGES_REQUESTED** [review]

File: scaleway/testdata/object-bucket-destroy-force.cassette.yaml [hunk 4]
```diff
@@ -0,0 +1,568 @@
...
+      Authorization:
+      - AWS4-HMAC-SHA256 Credential=SCWK721NRK9BFRHW1CS7/20220214/fr-par/s3/aws4_request,
+        SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=527fb8e17949cdd083ba828e464d51737333db1cee8dca364509b553f4aeb8a7
+      User-Agent:
+      - aws-sdk-go/1.42.48 (go1.17.6; linux; amd64)
+      X-Amz-Content-Sha256:
+      - e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+      X-Amz-Date:
+      - 20220214T140856Z
+    url: https://test-acc-scaleway-object-bucket-force-7443382734974225702.s3.fr-par.scw.cloud/
```
Comment (murlock) [inline scaleway/testdata/object-bucket-destroy-force.cassette.yaml:456]: Please use something like `https://test-acc-scaleway-object-bucket-force-7443382734974225702.s3.fr-par.scw.cloud/?versions=`
and response
```
<?xml version="1.0" encoding="UTF-8"?>\n<ListVersionsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Name>michael-versioning</Name><Prefix></Prefix><KeyMarker></KeyMarker><VersionIdMarker></VersionIdMarker><MaxKeys>1000</MaxKeys><EncodingType>url</EncodingType><IsTruncated>false</IsTruncated><DeleteMarker><Key>magic</Key><VersionId>uLkwWXneH5_gKvlqopQqKZaWhEUVY51m</VersionId><IsLatest>true</IsLatest><LastModified>2021-06-10T14:24:23.000Z</LastModified><Owner><ID>69a360b85b319b998292fbac13803cdc79898d1d2ed0f29f3a245874b0ebd97a</ID></Owner></DeleteMarker><Version><Key>magic</Key><VersionId>vrkAriN1FwpIibUSu9lH9VG65GYp28yB</VersionId><IsLatest>false</IsLatest><LastModified>2021-04-13T14:20:27.000Z</LastModified><ETag>&quot;d41d8cd98f00b204e9800998ecf8427e&quot;</ETag><Size>0</Size><Owner><ID>69a360b85b319b998292fbac13803cdc79898d1d2ed0f29f3a245874b0ebd97a</ID></Owner><StorageClass>STANDARD</StorageClass></Version></ListVersionsResult>
```

File: scaleway/testdata/object-bucket-destroy-force.cassette.yaml [hunk 5]
```diff
@@ -0,0 +1,568 @@
...
+      Authorization:
+      - AWS4-HMAC-SHA256 Credential=SCWK721NRK9BFRHW1CS7/20220214/fr-par/s3/aws4_request,
+        SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=a824271b984b609c4db1d9219e0e854febcbc1aea531f3e7aa3e1869f7755538
+      User-Agent:
+      - aws-sdk-go/1.42.48 (go1.17.6; linux; amd64)
+      X-Amz-Content-Sha256:
+      - e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+      X-Amz-Date:
+      - 20220214T140857Z
+    url: https://test-acc-scaleway-object-bucket-force-7443382734974225702.s3.fr-par.scw.cloud/test-file
```
Comment (murlock) [inline scaleway/testdata/object-bucket-destroy-force.cassette.yaml:520]: with versioning
`https://michael-versioning.s3.us-east-2.amazonaws.com/toto?versionId=LqBPq9jxroqe.ELbPRg8jq81ey29Zk8t`

**Review by Monitob — APPROVED** [review]

**Pull request merged**